core/src/com/unciv/ui/pickerscreens/ImprovementPickerScreen.kt

```diff
@@ -48,6 +48,7 @@ class ImprovementPickerScreen(val tileInfo: TileInfo, val onAccept: ()->Unit) :
         regularImprovements.defaults().pad(5f)
 
         for (improvement in tileInfo.tileMap.gameInfo.ruleSet.tileImprovements.values) {
+            if (improvement.hasUnique("Great Improvement")) continue
```

Comment:
It's the 'unbuildable' that should restrict it from being built, not the 'Great Improvement'

Reply:
Amazing to see another fix from you :D

Reply:
Done. I am currently a bit too busy to do big features, so some bug fixes from time to time is all I can do. ^_^